src/ast/passes/semantic_analyser.cpp

```diff
@@ -1464,29 +1464,73 @@ Probe *SemanticAnalyser::get_probe_from_scope(Scope *scope,
 void SemanticAnalyser::visit(Map &map)
 {
   MapKey key;
+  bool is_tuple_literal = false;
+
+  if (map.key_expr) {
+    Visit(map.key_expr);
+    Expression *expr = map.key_expr;
+
+    if (expr->type.IsTupleTy()) {
+      // We can only do this expression and size type manipulation
+      // if we have access to the underlying expressions in a tuple literal
+      // e.g. @map[(1, 2)] vs $a = (1, 2) @ma[$a]
+      is_tuple_literal = expr->is_literal;
+      if (is_tuple_literal) {
+        std::vector<SizedType> elements;
+        Tuple *tuple = static_cast<Tuple *>(map.key_expr);
+        // NOLINTBEGIN(modernize-loop-convert)
+        for (unsigned int i = 0; i < tuple->elems.size(); i++) {
+          // NOLINTEND(modernize-loop-convert)
+          Visit(tuple->elems.at(i));
+          Expression *elem = tuple->elems.at(i);
+
+          if (elem->type.IsNoneTy() || elem->type.GetSize() == 0)
+            return;
```

Comment:
Is this an error? And why the return?

Reply:
Added a comment which I borrowed from `visit(Tuple)`